### DISCUSSION TITLE
How to handle tests that need to reboot?

### DISCUSSION THREAD

**Discussion by silversocket** [pytest-dev/pytest#7818] [Q&A] [answered]
Is possible to reboot a box that is in the middle of a test, then start pytest back up, and pickup where it left off?
Or maybe reboot after a test, then pickup at the next test and continue?

EDIT: If it's not possible currently, if there was a $1million reward for supporting this functionality, at a high level, what changes would be required?

**Comment by The-Compiler** [comment, marked as answer]
No $1million reward needed, you could probably write a plugin doing this kind of thing. It would need to somehow keep track of which tests are already executed and write that to a file, and then use that to resume afterwards. Check pytest's own `--lf` or `--stepwise` for inspiration.

**Reply by silversocket** [reply]
That makes sense.
I'm trying to test out the ability to cache results as it runs, and when pytest runs again, I'd like all the previous tests to "rerun" except instead of _actually_ running the function, I just load data/results from the cache.
That way the terminal output and junit makes it look like everything ran without a reboot.
**Do you know how I can force load in the previous data per test?**
It looks like modifying `pytest_pyfunc_call` will allow me to run a stub function ✅ .
I tried using `pytest_runtest_logreport` to change the `report.outcome` to "failed" for an actually passing test, but in the end, the test still gets shown as passed in the final summary and the junit 🟥

**Reply by The-Compiler** [reply]
Maybe try marking your `pytest_runtest_logreport` as hookwrapper - see my (silly) [pytest-vw plugin](https://github.com/The-Compiler/pytest-vw/blob/master/pytest_vw.py#L29-L39) for inspiration.

**Reply by silversocket** [reply]
Excellent thanks! Looks like using `pytest_runtest_makereport` instead of `pytest_runtest_logreport` is working 🍾

**Reply by The-Compiler** [reply]
Ah, well spotted, I didn't even notice the difference when I was writing my comment  :laughing:

**Reply by nicoddemus** [reply]
@silversocket if this answered your question, could you mark it as answered then? Thanks!

**Comment by RonnyPfannschmidt** [comment]
my personal general suggestion is, that if a reboot control is needed, the test should be running on a system different from the one being reboot in order to orchestrate it in a manner that allows the bookkeeping to be done by a in test api

**Reply by silversocket** [reply]
I appreciate your suggestion. That is the common test architecture, but it would be really cool to be able to reboot on the same box.

**Reply by RonnyPfannschmidt** [reply]
if most programming languages and operating systems where in a position where a reboot mid process wouldn't loose state and still work reliable, i doubt we'd need reboots anymore to begin with ^^
